src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/RequestEvidenceBundleNotificationHandler.java

```diff
@@ -0,0 +1,127 @@
+    }
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && (callback.getEvent() == Event.REQUEST_RESPONDENT_EVIDENCE);
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        String homeOfficeReferenceNumber = asylumCase.read(HOME_OFFICE_REFERENCE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("Home office reference for the appeal is not present"));
+
+        final Optional<HomeOfficeCaseStatus> homeOfficeCaseStatus =
+            asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_CASE_STATUS_DATA, HomeOfficeCaseStatus.class);
+
+        if (homeOfficeCaseStatus.isPresent()) {
+            String documentReference = homeOfficeCaseStatus.get().getApplicationStatus().getDocumentReference();
+            if (!isEmpty(documentReference)) {
+                log.info("Using document reference {} from application status instead of home{}",
+                    documentReference, homeOfficeReferenceNumber);
+                homeOfficeReferenceNumber = documentReference;
+            }
+        }
+
+        final String caseId = asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("Case ID for the appeal is not present"));
+
+        final String correlationId = HomeOfficeRequestUuidGenerator.generateUuid();
+
+        PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
+
+        try {
+            final RequestEvidenceBundleInstructMessage bundleInstructMessage =
+                buildRequest(caseId, correlationId, homeOfficeReferenceNumber, asylumCase);
+
+            final HomeOfficeInstructResponse instructResponse =
+                homeOfficeInstructService.sendNotification(asylumCase, bundleInstructMessage);
+
+            if (instructResponse == null || instructResponse.getMessageHeader() == null) {
+                asylumCase.write(HOME_OFFICE_INSTRUCT_STATUS, "Internal Server Error");
+                log.error("Error sending notification to Home Office");
+            } else {
+                asylumCase.write(HOME_OFFICE_INSTRUCT_STATUS, "OK");
+            }
+        } catch (Exception e) {
+            log.error("Error sending notification to Home office. Message: " + e.getMessage());
+            asylumCase.write(
+                HOME_OFFICE_INSTRUCT_STATUS,
```

Comment:
We had earlier agreed to store only the HTTP status for HOME_OFFICE_INSTRUCT_STATUS. Check with Ravi if it is ok to store error message we need

Reply:
Agree..  Will make the changes. It will be either OK or FAIL similar to validation status. But this entire status thing could change in upcoming stories. But it is fine for the first one.